list/ConsoleApplication46/ConsoleApplication46.cpp

```diff
@@ -0,0 +1,115 @@
+﻿#include <iostream>
+#include<string>
+using namespace std;
+
+template<typename T>
+class Dick
+{
+public:
+	struct Node
+	{
+		T data;
+		Node* prev;
+		Node* next;
+
+		Node(T data)
+		{
+			this->data = data;
+			this->prev = this->next = NULL;
+		}
+	};
+	Node* head;
+	Node* tail;
+	Node size;
+public:
+	Dick()
+	{
+		this->head = this->tail = NULL;
+	}
+	Node* push_front(T data) 
+	{
+		Node* ptr = new Node(data);
+
+		ptr->next = head;
+		if (head != NULL)
+			head->prev = ptr;
+		if (tail == NULL)
+			tail = ptr;
+		head = ptr;
+
+		return ptr;
+	}
+
+
+	Node* push_back(T data) 
+	{
+		Node* ptr = new Node(data);
+
+		ptr->prev = tail;
+		if (tail != NULL)
+			tail->next = ptr;
+		if (head == NULL)
+			head = ptr;
+		tail = ptr;
+
+		return ptr;
+	}
+	int* pop_front()
+	{
+		if (head == NULL) return nullptr;
+
+		Node* ptr = head->next;
+		if (ptr != NULL)
+			ptr->prev = NULL;
+		else
+			tail = NULL;
+
+		delete head;
+		head = ptr;
+	}
+
+
+	Node* pop_back()
+	{
+		if (tail == NULL)
+		{ 
+			return NULL;
+		}
+			
+
+		Node* ptr = tail->prev;
+		if (ptr != NULL)
+			ptr->next = NULL;
+		else
+			head = NULL;
+
+		delete tail;
+		tail = ptr;
+	}
+	Node* find(T size)
```

Comment:
Why find function is getting "size"? The find function should get the index of the value. Also rename function please to "find_by_index"